backend/logger/fileHandler.py

```diff
@@ -0,0 +1,80 @@
+import os
+import uuid
+import datetime
+import atexit
+
+
+class FileHandler:
+    """
+    In charge of handling files for the logger. This class will open, close,
+    and name files for the logger. Responsible for storing logs in safe place.
+    In addition, it will pass file handles to different programs.
+    """
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.full_name = file_name
```

Comment:
Why do we assign the full name to just the file name, and then reassign it to the correct thing later? Is it possible to just call `getFullPath()` right off the bat?

Reply:
I will investigate this after writing the tests, it will be a bit tricky since they're involved in the logic of the code in other places.

Reply:
Any update on this? It looks like bugs could be caused if the full_name is not always the full name

Reply:
Will fix this.

Reply:
sounds good, lmk when done

Reply:
Hi @slakh96 , I looked at this again. There is actually nothing wrong with the logic. The file_name and full_name are both set to the file name when File Handler is created. Then the full name gets updated later down the line to include the UID and the date. The situation you've described above never happens because the file doesn't get created if full_name doesn't include UID and date.

Reply:
Any way to include the UID and date in the full_name to start off?

Reply:
That's not really possible. The whole point of all these functions is to create a unique name for each file. The full name and file name both exist and the fact that they are equal is used in the logic of the code later down the line when we check to see if we've already created a file. If file_name == full_name then the file has not been created yet.